adoc/chapters/programming_interface.adoc

```diff
@@ -6060,9 +6060,23 @@ to local memory on a <<device>>, and they define the *requirements* to memory
 objects which determine the scheduling of <<kernel,kernels>> (see
 <<sub.section.memmodel.app>>).
 
-A memory object requirement is created when an accessor is constructed, unless
-the accessor is a placeholder in which case the requirement is created when
-the accessor is bound to a <<command>> by calling [code]#handler::require()#.
+A memory object requirement is created by a <<command-group-function-object>>
+whenever a non-placeholder accessor is constructed during execution of that
+function and whenever a placeholder accessor is bound to a <<command>> during
+execution of that function by calling [code]#handler::require()#.  These
+requirements are created sometime after the <<command-group-function-object>>
+returns to the [code]#submit()# function and before the [code]#submit()#
+function itself returns.
+
+[NOTE]
+====
+Since the memory object requirements aren't created until after the command
+group function object returns, it is generally unwise to construct a buffer or
+image inside of command group scope.  The destructor for the buffer or image
+only blocks when there is a memory requirement, but variables defined in the
+command group function object go out of scope before those requirements are
```

Comment:
The problem I can see in my implementation is more like a deadlock.
But thinking more about it, I am unsure that not dealing with dependencies at accessor construction time instead of postponing after `submit` lambda execution can be a generic enough C++ implementation.

Reply:
It looks more like implementation detail & trade-off, so I can give up my previous comment.